roles/openshift_health_checker/openshift_checks/etcd_volume_check.py

```diff
@@ -0,0 +1,49 @@
+# pylint: disable=missing-docstring
+"""
+Ansible module for determining information about etcd volume size.
+"""
+
+from openshift_checks import OpenShiftCheck, get_var
+
+
+class EtcdVolumeCheck(OpenShiftCheck):
+    """Check that etcd volume capacity is not over 50%"""
+
+    name = "etcd_volume_check"
+    tags = ["etcd_volume_check", "etcd"]
+
+    def run(self, tmp, task_vars):
+        ansible_mounts = get_var(task_vars, "ansible_mounts")
+        etcd_mountpath = self.get_etcd_mountpath(ansible_mounts)
+        etcd_diskspace_avail = etcd_mountpath.get("size_available", 0)
+        etcd_totaldiskspace = etcd_mountpath.get("size_total", 0)
+
+        # max disk usage limit
+        diskspace_usage_lim = get_var(task_vars, "etcd_volume_usage_limit_bytes")
+        current_diskspace_usage = float(etcd_totaldiskspace - etcd_diskspace_avail)
+
+        if current_diskspace_usage > float(diskspace_usage_lim):
+            msg = "Current etcd volume usage ({actual} MB) is greater than the threshold limit of {limit} MB."\
+                .format(actual=self.to_megabytes(current_diskspace_usage), limit=self.to_megabytes(diskspace_usage_lim))
+            return {"failed": True, "msg": msg}
+
+        return {"changed": False}
+
+    @staticmethod
+    def get_etcd_mountpath(ansible_mounts):
+        # priority list - in descending priority order
+        valid_etcd_mount_paths = ["/var/lib/etcd", "/var/lib", "/var", "/"]
+
+        if not ansible_mounts:
+            return None
```

Comment:
By returning `None` here and in the last statement of this method, code that uses the return value can raise `TypeError` -- see `etcd_diskspace_avail = etcd_mountpath.get("size_available", 0)` above.